index/scorch/scorch.go

```diff
@@ -335,16 +335,19 @@ func (s *Scorch) Batch(batch *index.Batch) (err error) {
 		go func() {
 			for _, doc := range batch.IndexOps {
 				if doc != nil {
-					aw := index.NewAnalysisWork(s, doc, resultChan)
+					doc := doc
```

Comment:
So, this is the classic Go range variable capture issue.  The `doc` declared in the scope of the range statement is a single variable which changes each iteration of the loop.  Because doc is type interface, it's essentially a pointer, then these are passed via channels, and accessed later, by which time the contents of the pointer has changed to some later iteration.

The `doc := doc` is declaring a new variable inside the loop iteration, which copies the `doc` from the range statement.  This ensures the `doc` that is accessed later is a unique variable, not shared.

It is fundamentally the exact same as this line in upsidedown:

https://github.com/blevesearch/bleve/blob/c3457bb70309d45b3c5daef87d06f6a0b3d0813c/index/upsidedown/upsidedown.go#L824